### PR TITLE
Feature mixedreality

### PR DESCRIPTION
**Pull request by dsazulay**
Add Mixed Reality API to facilitate the development of AR programs.

This API expose to users the following functionalities:

* Detection and tracking of horizontal and vertical planes
* Creation and tracking of anchors
* Surface hit-testing to place contents
* Image detection and tracking
* Cloud anchor

In this first version the AR functionalities are being provided by ARCore only. The ideia is to add support to other AR frameworks like Vuforia in the future.

Demo of the API usage:
https://github.com/gearvrf/GearVRf-Demos/pull/619

GearVRf-DCO-1.0-Signed-off-by: Diego Azulay <diego.a@samsung.com>

### CODE REVIEW

**Comment by thomasflynn** [comment]
did an initial skim through this and the associated demo.  looks good so far.

**Comment by thomasflynn** [comment]
ok, did another pass. there's quite a few public classes, interfaces, and methods without javadoc. please correct that.

File: GVRf/Extensions/MixedReality/src/main/java/org/gearvrf/mixedreality/GVRMixedReality.java [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    public void setAugmentedImages(ArrayList<Bitmap> imagesList) {
+        mSession.setAugmentedImages(imagesList);
+    }
+
+    @Override
+    public ArrayList<GVRAugmentedImage> getAllAugmentedImages() {
+        return mSession.getAllAugmentedImages();
+    }
+
+    private class ActivityEventsHandler implements IActivityEvents {
```
Comment (liaxim) [inline GVRf/Extensions/MixedReality/src/main/java/org/gearvrf/mixedreality/GVRMixedReality.java]: GVREventListeners defines ActivityEvents - empty implementation of IActivityEvents. You could derive from that in the future.

**Comment by liaxim** [comment]
Looks good. Planning to merge on Wednesday.

**Pull request merged**